File: entsoe-client/src/main/java/org/example/model/load/LoadPeriod.java
package org.example.model.load;

import static org.example.model.load.GLMarketDocument.XML_NAMESPACE;

import java.util.List;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import lombok.Data;
import lombok.NoArgsConstructor;

// Period class for Load Domain with quantity points
@XmlAccessorType(XmlAccessType.FIELD)
@Data
@NoArgsConstructor
public class LoadPeriod {

  @XmlElement(name = "timeInterval", namespace = XML_NAMESPACE)
  private LoadTimeInterval timeInterval;

  @XmlElement(name = "resolution", namespace = XML_NAMESPACE)
  private String resolution;

  @XmlElement(name = "Point", namespace = XML_NAMESPACE)
  private List<QuantityPoint> points;
}

File: entsoe_client/src/entsoe_client/model/load/load_period.py
from pydantic_xml import BaseXmlModel, element

from entsoe_client.model import ENTSOE_LOAD_NSMAP
from entsoe_client.model.load.load_point import LoadPoint
from entsoe_client.model.load.load_time_interval import LoadTimeInterval


class LoadPeriod(BaseXmlModel, tag="Period", nsmap=ENTSOE_LOAD_NSMAP):  # type: ignore[call-arg]
    timeInterval: LoadTimeInterval = element(tag="timeInterval")
    resolution: str = element(tag="resolution")
    points: list[LoadPoint]
